x/accesscontrol/types/graph.go

```diff
@@ -130,6 +130,12 @@ func (dag *Dag) AddEdge(fromIndex DagNodeID, toIndex DagNodeID) *DagEdge {
 	if _, ok := dag.NodeMap[toIndex]; !ok {
 		return nil
 	}
+
+	// Processing in a given TX is sequential so no need for dependency
```

Comment:
why are we adding this check here? AddEdge is meant to be a helper that solely adds an edge, whereas helpers such as `AddNodeBuildDependency` will do the business logic of ensuring there arent edges within the same tx? I guess its fine to have this as an invariant for edge creation